lib/elixir/lib/range.ex

```diff
@@ -56,6 +56,41 @@ defmodule Range do
             "got: #{inspect(first)}..#{inspect(last)}"
   end
 
+  @doc """
+  Checks if two ranges overlap.
+
+  ## Examples
+
+      iex> Range.overlap?(1..7, 5..9)
+      true
+      iex> Range.overlap?(7..1, 5..9)
+      true
+      iex> Range.overlap?(1..7, 9..5)
+      true
+      iex> Range.overlap?(7..1, 9..5)
+      true
+      iex> Range.overlap?(0..1, 1..2)
+      true
+      iex> Range.overlap?(1..1, 1..1)
+      true
+      iex> Range.overlap?(1..5, 6..9)
+      false
+      iex> Range.overlap?(9..7, 3..1)
+      false
```

Comment:
Minor nitpicks: I would add `@doc since: "1.8.0"` and include a new line here :)

Reply:
Oh yes, pushed!